community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/fusion/FusionIndexProvider.java

```diff
@@ -163,8 +160,7 @@ public InternalIndexState getInitialState( long indexId, SchemaIndexDescriptor d
     @Override
     public IndexCapability getCapability( SchemaIndexDescriptor schemaIndexDescriptor )
     {
-        IndexCapability[] capabilities =
-                providers.instancesAs( new IndexCapability[INSTANCE_COUNT], provider -> provider.getCapability( schemaIndexDescriptor ) );
+        Iterable<IndexCapability> capabilities = providers.flatMap( provider -> provider.getCapability( schemaIndexDescriptor ) );
```

Comment:
When you rebase this the capabilities will no longer depend on the descriptor, so maybe you could store the union capability somewhere?

Reply:
Yeah, that might be a good idea!